linode/manager
# fix: [M3-8590] - Restrict access to Database Create page for restricted users

## Description 📝
Discovered during release testing for v1.128.0:

A recent PR updated the UI for users who have restricted access (read only or no access) to Databases: https://github.com/linode/manager/pull/10794

This included disabling buttons and providing helper tooltip text to explain the reason why restricted users couldn't access these features.

The PR did not include the same treatment for the databases/create page, which a restricted user can still navigate to via the URL (https://cloud.linode.com/databases/create) or via the top menu's Create button. The restricted user is able to fill out the form and submit it, and only then do they receive an error.

Thank you @mjac0bs for your inputs! :)

## Changes  🔄
List any change relevant to the reviewer.
- Added an error message on the create database page for restricted users
- Disabled all controls on /databases/create
- Updated DatabaseCreateAccessControls to disable inputs through a prop

## Target release date 🗓️
NA

## Preview 📷

![Screenshot 2024-10-23 at 4 29 20 PM](https://github.com/user-attachments/assets/a613ec03-619b-4550-bfec-312fe0ac81dc)

![Screenshot 2024-10-23 at 4 29 30 PM](https://github.com/user-attachments/assets/792ad739-2dc0-4076-b581-bb9208c80bf9)


## How to test 🧪


### Reproduction steps
1. Login as a user with full access and add databases. Grant read_only permissions to these new resources to the restricted user
2. Login as the restricted user and navigate to the databases page. The "Create Database Cluster" button should be disabled, but if you go to directly to https://cloud.linode.com/databases/create via the URL or via the top menu, notice that the form is accessible.

### Verification steps
Pull the PR, login as a restricted.
Navigate to /databases/create via the URL or via the top menu. The form should be disabled and an error notice should appear on top.

## As an Author I have considered 🤔

*Check all that apply*



---
